crucible-llvm/src/Lang/Crucible/LLVM/Translation.hs

```diff
@@ -369,9 +376,10 @@ insDeclareHandle halloc ctx decl = do
 -- Note: We may want to add a map from symbols to existing function handles
 -- if we want to support dynamic loading.
 translateModule :: HandleAllocator s -- ^ Generator for nonces.
+                -> (NonceGenerator (ST s) t)
                 -> L.Module          -- ^ Module to translate
-                -> ST s (Some ModuleTranslation)
-translateModule halloc m = do
+                -> ST s (Some (ModuleTranslation t))
+translateModule halloc nonceGen m = do
```

Comment:
I don't think we need a separate nonce generator here, you can get the generator out of the handle allocator using the record accessor `haCounter`.

Reply:
Ah, I missed that because I saw that the constructor wasn't exported. Thanks!